libcloud/compute/drivers/gridscale.py

```diff
@@ -0,0 +1,977 @@
+        return result.status == 204
+
+    def delete_image(self, node_image):
+        """
+        Destroy an image.
+
+        :param node_image: Node image object.
+        :type node_image: :class:`.NodeImage`
+
+        :return: True if the destroy was successful, otherwise False
+        :rtype: ``bool``
+
+        """
+        result = self._sync_request(endpoint='objects/templates/{}'
+                                    .format(node_image.id),
+                                    method='DELETE')
+        return result.status == 204
+
+    def patch_node(self, node, name=None):
+        """
+        change node name
+
+        :param name: new name
+        :type name: ``str``
+
+        :param node: Server you want to patch
+        :type node: ``str``
+
+        :return: ``True`` or ``False``
+        :rtype: ``bool``
+        """
+        result = self._sync_request(data=json.dumps({'name': name}),
+                                    endpoint='objects/servers/{}'
+                                    .format(node.id),
+                                    method='PATCH')
+        return result.status == 204
+
+    def patch_storage(self, storage, name=None):
+        """
+        modify storage name
+
+        :param storage: storage object
+        :type storage: ``object``
+
+        :param name: rename storage
+        :type name: ``str``
+
+        :return: ``True`` or ``False``
+        :rtype: ``bool``
+        """
+        result = self._sync_request(data=json.dumps({'name': name}),
+                                    endpoint='objects/servers/{}'
+                                    .format(storage.id),
+                                    method='PATCH')
+        return result.status == 204
+
+    def ex_patch_networks(self, network, name=None):
+        """
+        Modify networks name.
+
+        :param network: Network object.
+        :type network: :class:`.GridscaleNetwork`
+
+        :param name: New name.
+        :type name: ``str``
+
+        :return: ``True`` or ``False``
+        :rtype: ``bool``
+        """
+        result = self._sync_request(data=json.dumps({'name': name}),
+                                    endpoint='objects/networks/{}'
+                                    .format(network.id),
+                                    method='PATCH')
+        return result.status == 204
+
+    def reboot_node(self, node):
+        """
+        Reboot a node.
+
+        :param node: Node object.
+        :type node: :class:`.Node`
+
+        :return: True if the reboot was successful, otherwise False.
+        :rtype: ``bool``
+
+        """
+
+        if node.extra['power'] is True:
+            data = dict({'power': False})
+            self._sync_request(data=json.dumps(data),
+                               endpoint='objects/servers/{}/power'
+                               .format(node.id),
+                               method='PATCH')
+            time.sleep(3)
```

Comment:
Perhaps we should make this value user-configurable - e.g. ``ex_sleep_interval=3``.

Also, I assume this perform a "hard" power off and not a soft one? For a soft one, I would imagine 3 seconds is way too low.